core/tests/unittests/metrics/test_metrics.py

```diff
@@ -140,6 +152,60 @@ def test_metrics_perfect_multiclass(metric: str):
     _assert_perfect_score(scorer=METRICS[MULTICLASS][metric])
 
 
+@pytest.mark.skip(reason="average_precision doesn't raise an exception here when it should, so this test currently fails")
+@pytest.mark.parametrize("metric", METRICS_NEEDS_THRESHOLD[BINARY], ids=METRICS_NEEDS_THRESHOLD[BINARY])  # noqa
+def test_metrics_perfect_raises_binary_single_sample(metric: str):
+    with pytest.raises(Exception):
+        # threshold metrics should not be able to predict on a single sample
+        _assert_perfect_score_single_sample(scorer=METRICS[BINARY][metric])
+
+
+@pytest.mark.skip(reason="mcc, quadradic_kappa, and pac produce unexpected values here, so this test currently fails")
```

Comment:
Do we plan to fix these tests in a follow up PR?

Reply:
Possibly, however as in the prior comment on average_precision the details are quite nuanced for each, while the scenarios this comes up are very niche.